File: erlang/06-io-operations/read_file.erl
#!/usr/bin/env escript
%% read_file.erl - Reading data back from a file
main(_) ->
    {ok, Bin} = file:read_file("languages.txt"),
    io:format("--- File contents ---~n~s", [Bin]),

    %% Split on newlines to count the lines (trim the trailing newline first)
    Lines = string:split(string:trim(Bin), "\n", all),
    io:format("Line count: ~B~n", [length(Lines)]).

File: php/06-io-operations/read_file.php
<?php
// Create a sample file so this example is self-contained
file_put_contents("sample.txt", "apple\nbanana\ncherry\n");

// Read the whole file into one string
$contents = file_get_contents("sample.txt");
echo "Full contents:\n";
echo $contents;

// Read the file into an array of lines, dropping the newlines
$lines = file("sample.txt", FILE_IGNORE_NEW_LINES);
echo "Line count: " . count($lines) . "\n";

// Read line by line with a stream — ideal for large files
echo "Numbered lines:\n";
$handle = fopen("sample.txt", "r");
$number = 1;
while (($line = fgets($handle)) !== false) {
    echo $number . ": " . trim($line) . "\n";
    $number++;
}
fclose($handle);
